dipy/viz/tests/test_ui.py

```diff
@@ -0,0 +1,131 @@
+import os
+from collections import defaultdict
+
+from os.path import join as pjoin
+
+import numpy.testing as npt
+
+from dipy.data import read_viz_icons, fetch_viz_icons
+from dipy.viz import interactor
+from dipy.viz import ui
+from dipy.viz import window
+from dipy.data import DATA_DIR
+
+from dipy.testing.decorators import xvfb_it
+
+# Conditional import machinery for vtk
+from dipy.utils.optpkg import optional_package
+
+# Allow import, but disable doctests if we don't have vtk
+from dipy.viz.ui import UI
+
+vtk, have_vtk, setup_module = optional_package('vtk')
+
+use_xvfb = os.environ.get('TEST_WITH_XVFB', False)
+if use_xvfb == 'skip':
+    skip_it = True
+else:
+    skip_it = False
+
+
+@npt.dec.skipif(not have_vtk or skip_it)
+@xvfb_it
+def test_button(recording=False):
+    print("Using VTK {}".format(vtk.vtkVersion.GetVTKVersion()))
+    filename = "test_ui.log.gz"
+    recording_filename = pjoin(DATA_DIR, filename)
+    renderer = window.ren()
+
+    # Define some counter callback.
+    states = defaultdict(lambda: 0)
+
+    # Broken UI Element
+    class BrokenUI(UI):
+
+        def __init__(self):
+            super(BrokenUI, self).__init__()
+
+        def get_actors(self):
+            pass
+
+        def set_center(self, position):
+            pass
+
+    broken_ui = BrokenUI()
+    npt.assert_raises(NotImplementedError, broken_ui.get_actors())
```

Comment:
This doesn't raise an error for me. Is it because I actually implemented a 

    def foo(self):
         pass

for all the unimplemented functions? If I don't they throw an error during initialisation. How do I prevent that?